marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/flows/test-data/valueStep.sjs

```diff
@@ -0,0 +1,35 @@
+const DataHubSingleton = require("/data-hub/5/datahub-singleton.sjs");
+const datahub = DataHubSingleton.instance();
+const expectedValues = "comma-separatedurisofrecordstoprocess";
+//Here we should receive a sequence of values, not a content object
+function main(values, options) {
+  let concatString = "";
+  //check to see how many values we have, we should have more than 1
+  if(fn.count(values) < 2) {
+    throw Error("Fewer than expected values passed in");
+  }
+  //we should be able to loop through the list of values and identify them as strings
+  for(let value of values){
+    //normalize it to the value from a value object
+    let valueNormalized = value.valueOf();
+    //now we check to make sure its of type string, as it should be from the uri lexicon
+    //if not, toss an error
+    if(!valueNormalized instanceof String){
+      let errMsg = valueNormalized + ' is not a string value.';
+      datahub.debug.log({message: errMsg, type: 'error'});
+      throw Error(errMsg);
+    } else {
+    concatString += valueNormalized;
+    }
+  }
+  if(concatString !=  expectedValues) {
+          datahub.debug.log({message: "Expected concat of values did not match.", type: 'error'});
+          throw Error("Expected concat of values did not match. Got: " +concatString + " but was expecting:" + expectedValues);
+  }
+
+  return null;
```

Comment:
What do you think of returning a content object here that has a document in it containing the values? The test can then verify that the document was inserted with the expected values from the values query.

Reply:
I initially did that, but the order of the values is important to be preserved (in the order they were submitted) as they are processed (same with an acceptsBatch sequence) due to the ordered nature of sequence (esp if it is sorted by order in the batching already).

Could probably do another test with a list of values that are integer based and pass them in as a batch, being able to cleanly loop over and extract the values from them and compare them with an expected to cover ourselves both ways.